spec/system/consumer/shopping/cart_spec.rb

```diff
@@ -269,6 +269,21 @@
           end
         end
       end
+
+      describe "when user changes number in text input, check price" do
+        it "allows updating the quantity" do
+          variant.update!(on_hand: 5, on_demand: true)
+
+          visit main_app.enterprise_shop_path(distributor)
+
+          fill_in "variant_line_item_quantity_#{variant.id}", with: '3'
```

Comment:
You use the variant id here even though the view uses the line item id. But I made a suggestion above which uses the variant id as well.